### ISSUE TITLE
SPSecureStoreServiceApp ignores DatabaseName property

### ISSUE THREAD

**Issue by VerdonTrigance** [dsccommunity/SharePointDsc#1263]
### Description

```
SPSecureStoreServiceApp SecureStoreServiceApp
        {
            Name                  = "Secure Store Service Application"
            ProxyName = "Secure Store Service Application Proxy"
            ApplicationPool       = $serviceAppPoolName
            AuditingEnabled       = $true
            AuditlogMaxSize       = 30
            DatabaseName          = "$($Node.DBPrefix)_SecureStoreService"
            DatabaseServer = $Node.DatabaseServer
            PsDscRunAsCredential  = $SPSetupAccount
            DependsOn             = "[SPServiceAppPool]MainServiceAppPool"
        }
```
Results to creating database with name 'Secure_Store_Service_DB_5abb799c-fff4-4cad-9b6d-c5033280e51b'

Log:

```
VERBOSE: [dev-02]: LCM:  [ Start  Resource ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]
VERBOSE: [dev-02]: LCM:  [ Start  Test     ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Testing secure store service application Secure Store Service Application
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Getting secure store service application 'Secure Store Service Application'
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Executing as the local run as user dom\admin
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving BeginProcessing Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving ProcessRecord Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving EndProcessing Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Current Values: ApplicationPool=SharePoint Service Applications; AuditingEnabled=False; Ensure=Absent; Name=Secure Store Service Application
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Target Values: ApplicationPool=SharePoint Service Applications; AuditingEnabled=True; AuditlogMaxSize=30; DatabaseName=dev02_SecureStoreService; Databa
seServer=dev-sql; Ensure=Present; Name=Secure Store Service Application; ProxyName=Secure Store Service Application Proxy; Verbose=True
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Test-TargetResource returned False
VERBOSE: [dev-02]: LCM:  [ End    Test     ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]  in 0.6250 seconds.
VERBOSE: [dev-02]: LCM:  [ Start  Set      ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Setting secure store service application 'Secure Store Service Application'
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Getting secure store service application 'Secure Store Service Application'
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Executing as the local run as user dom\admin
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving BeginProcessing Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving ProcessRecord Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving EndProcessing Method of Get-SPServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Creating Secure Store Service Application Secure Store Service Application
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Executing as the local run as user dom\admin
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving BeginProcessing Method of New-SPSecureStoreServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving BeginProcessing Method of New-SPSecureStoreServiceApplicationProxy.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] The proxy to the service application was added.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving ProcessRecord Method of New-SPSecureStoreServiceApplicationProxy.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving ProcessRecord Method of New-SPSecureStoreServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving EndProcessing Method of New-SPSecureStoreServiceApplication.
VERBOSE: [dev-02]:                            [[SPSecureStoreServiceApp]SecureStoreServiceApp] Leaving EndProcessing Method of New-SPSecureStoreServiceApplicationProxy.
VERBOSE: [dev-02]: LCM:  [ End    Set      ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]  in 37.1210 seconds.
VERBOSE: [dev-02]: LCM:  [ End    Resource ]  [[SPSecureStoreServiceApp]SecureStoreServiceApp]
```

PowerShell DSC 4.3.0
SharePoint Server 2019

**Comment by ykuijs** [comment]
Hi @VerdonTrigance, thanks for reporting! This issue was reported earlier as well (#1249) and is fixed in v4.4. Please try that version to see if the issue is resolved on your environment.

**Comment by VerdonTrigance** [comment]
> Hi @VerdonTrigance, thanks for reporting! This issue was reported earlier as well (#1249) and is fixed in v4.4. Please try that version to see if the issue is resolved on your environment.

4.4.0 works well. Thank you.
